compiler/AST/FnSymbol.cpp

```diff
@@ -1152,9 +1139,19 @@ const char* toString(FnSymbol* fn) {
       retval     = astr(retval, "(");
     }
 
-    for (int i = start; i <= fn->numFormals(); i++) {
+    for (int i = 1; i <= fn->numFormals(); i++) {
       ArgSymbol* arg = fn->getFormal(i);
 
+      // skip method token etc
+      if (arg->type == dtMethodToken ||
+          arg->type == dtTypeDefaultToken ||
+          arg->type == dtModuleToken)
+        continue;
+
+      // skip this formal methods in non-developer mode
```

Comment:
Could you please clarify the above comment.

Maybe add a brief note explaining why skip `this` in non-devel mode?